File: 198.py
# 198: House Robber
# Tested successfully



nums1 = [1, 2, 3, 1]
nums2 = [2, 7, 9, 3, 1]

def maxRobbingAmount(nums):
    arr = [nums[0], nums[1]]
    for i in range(2, len(nums)):
        arr +=  [max(arr[i-2] + nums[i], arr[i-1])]

    return arr


print(maxRobbingAmount(nums1))

File: 198.js
// 198: House Robber
// Tested successfully


let nums1 = [1, 2, 3, 1];
let nums2 = [2, 7, 9, 3, 1];

function houseRobber(nums)
{
    let ans = [nums[0], nums[1]];
    for (let i = 2; i < nums.length; i++)
    {
        ans.push(Math.max(ans[i-2] + nums[i], ans[i-1]));
    }
    return ans.pop();
}

